File: AppliSportSpringBoot/src/main/java/projet/AppliSport/model/Profil.java
package projet.AppliSport.model;

import java.time.LocalDate;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;
import javax.validation.constraints.DecimalMax;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Past;

import org.springframework.format.annotation.DateTimeFormat;



@Entity
@Table(name="profil")
@SequenceGenerator(name="seqProfil", sequenceName = "seq_profil",initialValue = 100, allocationSize = 1)
public class Profil {

	@Id
	@GeneratedValue (strategy = GenerationType.SEQUENCE,generator = "seqProfil")
	@Column(name="profil_id")
	private Long id;
	
	@DecimalMin(value = "0")
	@Column(name="profil_poids")
	private double poids;
	
	@DecimalMin(value = "0.5")
	@DecimalMax(value = "3.0")
	@Column(name="profil_taille")
	private double taille;
	
	@Past
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@Column(name="profil_naissance")
	private LocalDate dateNaissance;
	
	@NotEmpty
	@Column(name="profil_sexe")
	@Enumerated (EnumType.STRING)
	private Sexe sexe;
	
	@OneToOne
	@JoinColumn(name = "profil_caracteristique", foreignKey = @ForeignKey(name = "profil_caracteristique_fk"))
	private Caracteristique caracteristique;
	//@OneToMany(mappedBy = "profilSport")
	//private Set<Sport> sports;
	@OneToOne(mappedBy = "profilUtilisateur")
	private Utilisateur utilisateur;
	
	@Version
	private int version;
	
	public Profil() {
		
	}
	
	

	
	
	public Profil(Long id, double poids, double taille, LocalDate dateNaissance, Sexe sexe,
			Caracteristique caracteristique) {
		this.id = id;
		this.poids = poids;
		this.taille = taille;
		this.dateNaissance = dateNaissance;
		this.sexe = sexe;
		this.caracteristique = caracteristique;

	}





	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public double getPoids() {
		return poids;
	}

	public double getTaille() {
		return taille;
	}

	public LocalDate getDateNaissance() {
		return dateNaissance;
	}

	public Sexe getSexe() {
		return sexe;
	}

	

	public void setPoids(double poids) {
		this.poids = poids;
	}

	public void setTaille(double taille) {
		this.taille = taille;
	}

	public void setDateNaissance(LocalDate dateNaissance) {
		this.dateNaissance = dateNaissance;
	}

	public void setSexe(Sexe sexe) {
		this.sexe = sexe;
	}

	



	public Caracteristique getCaracteristique() {
		return caracteristique;
	}





	public void setCaracteristique(Caracteristique caracteristique) {
		this.caracteristique = caracteristique;
	}



	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Profil other = (Profil) obj;
		return Objects.equals(id, other.id);
	}
	
	
	
}

File: SportifyApp/src/app/model/profil.ts
import { Caracteristique } from './caracteristique';
import { Sexe } from './sexe';
import { Utilisateur } from './utilisateur';

export class Profil {
  private _id: number | undefined;
  private _poids: number | undefined;
  private _taille: number | undefined;
  private _dateNaissance: Date | undefined;
  private _sexe: Sexe | undefined;
  private _caracteristique: Caracteristique | undefined;
  private _utilisateur: Utilisateur | undefined;

  constructor(
    id?: number,
    poids?: number,
    taille?: number,
    dateNaissance?: Date,
    sexe?: Sexe,
    caracteristique?: Caracteristique,
    utilisateur?: Utilisateur
  ) {
    this._id = id;
    this._poids = poids;
    this._taille = taille;
    this._dateNaissance = dateNaissance;
    this._sexe = sexe;
    this._caracteristique = caracteristique;
    this._utilisateur = utilisateur;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter poids
   * @return {number }
   */
  public get poids(): number | undefined {
    return this._poids;
  }

  /**
   * Getter taille
   * @return {number }
   */
  public get taille(): number | undefined {
    return this._taille;
  }

  /**
   * Getter dateNaissance
   * @return {Date }
   */
  public get dateNaissance(): Date | undefined {
    return this._dateNaissance;
  }

  /**
   * Getter sexe
   * @return {Sexe }
   */
  public get sexe(): Sexe | undefined {
    return this._sexe;
  }

  /**
   * Getter caracteristique
   * @return {Caracteristique }
   */
  public get caracteristique(): Caracteristique | undefined {
    return this._caracteristique;
  }

  /**
   * Getter utilisateur
   * @return {Utilisateur }
   */
  public get utilisateur(): Utilisateur | undefined {
    return this._utilisateur;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter poids
   * @param {number } value
   */
  public set poids(value: number | undefined) {
    this._poids = value;
  }

  /**
   * Setter taille
   * @param {number } value
   */
  public set taille(value: number | undefined) {
    this._taille = value;
  }

  /**
   * Setter dateNaissance
   * @param {Date } value
   */
  public set dateNaissance(value: Date | undefined) {
    this._dateNaissance = value;
  }

  /**
   * Setter sexe
   * @param {Sexe } value
   */
  public set sexe(value: Sexe | undefined) {
    this._sexe = value;
  }

  /**
   * Setter caracteristique
   * @param {Caracteristique } value
   */
  public set caracteristique(value: Caracteristique | undefined) {
    this._caracteristique = value;
  }

  /**
   * Setter utilisateur
   * @param {Utilisateur } value
   */
  public set utilisateur(value: Utilisateur | undefined) {
    this._utilisateur = value;
  }
}
